cellarium/ml/module/base_module.py

```diff
@@ -24,17 +30,38 @@ def _get_fn_args_from_batch(tensor_dict: dict[str, np.ndarray | torch.Tensor]) -
         Get forward method arguments from batch.
         """
 
+    def __getattr__(self, name: str) -> Any:
```

Comment:
It seems like you have brought in some parts of the `PyroModule` code into `BaseModule` and got rid of `BasePyroModule`. What's the logic?

Reply:
I wanted to have a single base model both for regular and pyro models. The only feature needed from `PyroModule` is the ability to have constrained `PyroParam`s. `PyroModule` also has a code that syncs parameters with `ParamStoreDict`. This is the stripped down version of `PyroModule` that only handles constrained params.